inst/examples/variable.cpp

```diff
@@ -38,6 +38,25 @@ class variable {
   dust::shared_ptr<variable> shared;
 };
 
+// This is required to activate gpu support. We can possibly do this
+// automatically based on the existance of the template below though.
+template <>
+struct dust::has_gpu_support<variable> : std::true_type {};
+
+template <>
+void update_device<variable>(size_t step,
+                             const dust::interleaved<variable::real_t> state,
+                             dust::interleaved<int> internal_int,
+                             dust::interleaved<variable::real_t> internal_real,
+                             dust::shared_ptr<variable> shared,
+                             dust::rng_state_t<variable::real_t>& rng_state,
+                             dust::interleaved<variable::real_t> state_next) {
+  for (size_t i = 0; i < shared->len; ++i) {
+    state_next[i] =
+      dust::distr::rnorm(rng_state, state[i] + shared->mean, shared->sd);
```

Comment:
These shared members access would then work the same way as the unpacking of internal state from before

Reply:
Given the mess above, it would be nice if we could use a pointer to a struct even if that struct contains just int, real_t, int* and real_t* (no vectors) with the allocation handled elsewhere. Otherwise the unpack at the beginning of the function is going to be quite painful